Jenkinsfile

```diff
@@ -1,216 +1,123 @@
 // vim: ft=groovy
 
-// Helpers for setting commit status
-def getRepoUrl() {
-    return sh(script: "git config --get remote.origin.url", returnStdout: true).trim()
-}
-
-def getCommitSha() {
-    return sh(script: "git rev-parse HEAD", returnStdout: true).trim()
-}
-
-
-def setGitHubBuildStatus(status) {
-    def repoUrl = getRepoUrl()
-    def commitSha = getCommitSha()
-
-    step([
-        $class: 'GitHubCommitStatusSetter',
-        reposSource: [$class: "ManuallyEnteredRepositorySource", url: repoUrl],
-        commitShaSource: [$class: "ManuallyEnteredShaSource", sha: commitSha],
-        errorHandlers: [[$class: 'ShallowAnyErrorHandler']],
-        statusResultSource: [
-          $class: 'ConditionalStatusResultSource',
-          results: [
-            [$class: 'BetterThanOrEqualBuildResult', result: 'SUCCESS', state: 'SUCCESS', message: status],
-            [$class: 'BetterThanOrEqualBuildResult', result: 'FAILURE', state: 'FAILURE', message: status],
-            [$class: 'AnyBuildResult', state: 'FAILURE', message: 'Loophole']
-          ]
-        ]
-    ])
-}
+def NUM_CORES = 8
+def errors = []
+def labels = ['CentOS', 'Fedora', 'Ubuntu', 'Mac'] // labels for Jenkins node types we will build on
+def nodes = [:] 
+
+for (lbl in labels) {
+    def label = lbl
+    def envFile = (label == "CentOS") ? "environment_gcc4.yml" : "environment.yml"
+
+    nodes[label] = {
+        stage(label) {
+            isisNode(label) {
+                def isisEnv = [
+                    "ISIS3DATA=/isisData/data",
+                    "ISIS3TESTDATA=/isisData/testData",
+                    "ISIS3MGRSCRIPTS=/isisData/data/isis3mgr_scripts",
+                    "MALLOC_CHECK_=1"
+                ]
+
+                def cmakeFlags = [
+                    "-DJP2KFLAG=ON",
+                    "-DKAKADU_INCLUDE_DIR=/isisData/kakadu",
+                    "-Dpybindings=OFF",
+                    "-DCMAKE_BUILD_TYPE=RELEASE"
+                ]
+
+                def stageStatus = "Checking out ISIS on ${label}"
+
+                // Checkout
+                checkout scm
+
+                if (label == "Mac") {
```

Comment:
> any significant difference between loginshell and sh? I would assume login shell to change depending on the default login shell in the container/mac machine rather than always using sh? In which case is there any reason we wouldn't want to force bash?

Added here